osbenchmark/aggregator.py

```diff
@@ -13,23 +13,31 @@ def __init__(self, cfg, test_executions_dict, args):
         self.args = args
         self.test_executions = test_executions_dict
         self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
-        self.accumulated_iterations: Dict[str, int] = {}
+        self.accumulated_iterations: Dict[str, Dict[str, int]] = {}
         self.metrics = ["throughput", "latency", "service_time", "client_processing_time", "processing_time", "error_rate", "duration"]
         self.test_store = metrics.test_execution_store(self.config)
         self.cwd = cfg.opts("node", "benchmark.cwd")
 
-    def count_iterations_for_each_op(self) -> None:
+    def count_iterations_for_each_op(self, test_execution) -> None:
         loaded_workload = workload.load_workload(self.config)
         test_procedure_name = self.config.opts("workload", "test_procedure.name")
```

Comment:
Should we initialize this as an instance variable so that it can be used throughout the aggregator class? Seeing areas where the aggregator has to fetch the test procedure name.

Reply:
Yeah, and for clarity I renamed the test_procedure in the build_aggregated_results function since it's not just the string name of the test procedure, but a test_procedure object with other attributes that are needed there